src/Database/Schema/PostgresSchemaDialect.php

```diff
@@ -431,21 +433,21 @@ public function columnSql(TableSchema $schema, string $name): string
             TableSchemaInterface::TYPE_JSON => ' JSONB',
         ];
 
-        if (isset($typeMap[$data['type']])) {
-            $out .= $typeMap[$data['type']];
-        }
-
-        $isInt = [
+        $autoIncrementTypes = [
+            TableSchemaInterface::TYPE_TINYINTEGER,
+            TableSchemaInterface::TYPE_SMALLINTEGER,
             TableSchemaInterface::TYPE_INTEGER,
             TableSchemaInterface::TYPE_BIGINTEGER,
         ];
-        if (in_array($data['type'], $isInt, true)) {
-            $type = $data['type'] === TableSchemaInterface::TYPE_INTEGER ? ' INTEGER' : ' BIGINT';
-            if ($schema->getPrimaryKey() === [$name] || $data['autoIncrement'] === true) {
-                $type = $data['type'] === TableSchemaInterface::TYPE_INTEGER ? ' SERIAL' : ' BIGSERIAL';
-                unset($data['null'], $data['default']);
+        if (in_array($data['type'], $autoIncrementTypes, true)) {
+            if (($schema->getPrimaryKey() === [$name] && $name === 'id') || $data['autoIncrement']) {
+                $typeMap[$data['type']] = str_replace('INT', 'SERIAL', $typeMap[$data['type']]);
```

Comment:
There isn't a `SMALLSERIAL` type in postgres so this will likely fail should someone try to do that.

Reply:
`SMALLSERIAL` is actually a legit type. https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-SERIAL

I've never used it before though.

Reply:
Darn, I missed that I was looking at the postgres 9 docs earlier :cry: